python/lsst/dax/distribution/DataIngest.py

```diff
@@ -0,0 +1,297 @@
+        """
+        url = self._base_url + ingest_cmd
+        print('url=', url, " data=", data_json)
+        response = requests.put(url, json=data_json)
+        status_code = response.status_code
+        success = True
+        # 200 means the put request was at least well formed.
+        if status_code != 200:
+            print('ERROR put url=', url, "data=", data_json, 'status=', status_code)
+            success = False
+            return success, status_code, None
+        r_json = response.json()
+        if not r_json['success']:
+            print('ERROR put url=', url, 'status=', status_code, 'data=', data_json,
+                  'r_json=', r_json)
+            success = False
+        return success, status_code, r_json
+
+    def sendDatabase(self, db_file_path):
+        """ Send the database description to the ingest system.
+        """
+        with open(db_file_path, 'r') as db_f:
+            data = db_f.read()
+            data_json = json.loads(data)
+        success, r_json = self._postToIngest('ingest/database', data_json)
+        if not success:
+            print('ERROR while sending databaae ', db_file_path, "r_json=", r_json)
+            return False
+        return True
+
+    def sendTableSchema(self, schema_file_path):
+        with open(schema_file_path, 'r') as schema_f:
+            data = schema_f.read()
+            data_json = json.loads(data)
+        success, r_json = self._postToIngest('ingest/table', data_json)
+        if not success:
+            print('ERROR while sending table schema ', schema_file_path, "r_json=", r_json)
+            return False
+        return True
+
+    def startTransaction(self, db_name):
+        """ Start an ingest super transaction.
+
+        Return
+        ------
+        success : bool
+            True when transaction was started successfully.
+        id : int
+            Ingest super transaction id number.
+        """
+        success, r_json = self._postToIngest('ingest/trans', {'database':db_name,'auth_key':''})
+        if not success:
+            print('ERROR when starting transaction ', db_name, "r_json=", r_json)
+            return False, -1
+        id = r_json['databases'][db_name]['transactions'][0]['id']
+        print("transaction id=", id, " r_json=", r_json)
+        return True, id
+
+    def endTransaction(self, transaction_id, abort):
+        """ End the ingest transaction
+
+        Parameters
+        ----------
+        transaction_id : int
+            The transaction id number.
+        abort : bool
+            True if the transaction should be rolled back.
+
+        Return
+        ------
+        success : bool
+            True if the operation was successful.
+        status : int
+            Status code from the put operation.
+        r_json : json or None
+            json data from the put operation if status was 200.
+        """
+        cmd = 'ingest/trans/' + str(transaction_id) + '?abort='
```

Comment:
You may rewrite this as a single line in Python 2:
```
cmd = 'ingest/trans/%d?abort=%d' % (transaction_id,abort,)
```
Or, like this in Python 3:
```
cmd = 'ingest/trans/{}?abort={}'.format(transaction_id,abort,)
```

Reply:
Both of the following return "ingest/trans/543?abort=True" which doesn't match a 0 or 1 value for abort.
cmd = 'ingest/trans/{}?abort={}'.format(transaction_id,abort)
cmd = f'ingest/trans/{transaction_id}?abort={abort}'

This does work
cmd = 'ingest/trans/%d?abort=%d' % (transaction_id,abort)

Reply:
My bad. I have corrected my original suggestion into:
```
cmd = 'ingest/trans/%d?abort=%d' % (transaction_id,int(abort),)
cmd = 'ingest/trans/{}?abort={}'.format(transaction_id,int(abort),)
```
This should work.